mlir/lib/Dialect/Affine/Utils/Utils.cpp

```diff
@@ -93,8 +95,9 @@ class AffineApplyExpander
     Value zeroCst = builder.create<arith::ConstantIndexOp>(loc, 0);
     Value isRemainderNegative = builder.create<arith::CmpIOp>(
         loc, arith::CmpIPredicate::slt, remainder, zeroCst);
-    Value correctedRemainder =
-        builder.create<arith::AddIOp>(loc, remainder, rhs);
+    Value correctedRemainder = builder.create<arith::AddIOp>(
```

Comment:
I'm pretty sure this addition is not `nuw`.

Specifically, consider `lhs = -5` and `rhs = 3`. Then, `remainder == -2`, which means that we're doing `correctedRemainder = add nuw nsw -2, 3`, which is not `nuw`